### DISCUSSION TITLE
What is the future of netlify-cms?

### DISCUSSION THREAD

**Discussion by bmackinney** [decaporg/decap-cms#6503] [Q&A] [answered]
WIth @erezrokah and previously @erquhart having left Netlify, progress seems to have halted here. Does Netlify have a plan for moving forward? Who is reviewing pull requests?

edit: see https://github.com/netlify/netlify-cms/graphs/code-frequency

**Comment by erezrokah** [comment]
Thanks for opening the discussion @bmackinney, tagging @krider2010

**Comment by bmackinney** [comment]
No plan, no one is reviewing. It's time for a fork.

**Comment by talves** [comment]
Moving this here for reference: https://answers.netlify.com/t/is-this-project-dead/70988

@bmackinney thanks for asking the question.

**Comment by CamilleScholtz** [comment]
Considering there is no real answer, it might indeed be time for a fork. Problem is of course is that we need someone dedicated enough for this.

**Reply by talves** [reply]
This runs into the issue of sustainability vs the trade-offs.

**tldr:** In my opinion, `netlify-cms` is not savable in the state it is right now. A fork would only perpetuate the errors of complexity it's tried to solve that should have been outside the CMS. I believe there are some really good pieces of `netlify-cms` that could be used, but a change of scope to keep it simple would be much better.

- Most large companies with the financial ability to maintain a project like this could easily justify the cost of a SaaS CMS solution vs the cost of trying to maintain a full CMS solution. Actually this cost could even be justified by medium sized companies.

- There are plenty of CMS solutions that have additional features this project has fallen short on but are too much for simple entry solutions for the personal projects of individuals and small projects (customizable style, better UI/UX, user roles, i8n, directory structure, clearly defined api, etc.)

- To be able to maintain a project of this size, it takes at least one full time maintainer and a few trusted maintainers that have publish access. Also a lot of dedicated contributors. With the right maintainers, a project can be successful. With a single maintainer keeping control of publishing and scope, it will fail eventually. Especially when they leave and have no more incentive in the project as we've seen here.

I've watched this project aim to solve trying to be a drop in solution for simple projects while also trying to solve some perceived issue for  complex projects and watched it fail over and over on that front. Using git to drive simple data structure for publishing is a good model using markdown and was a noble idea in the beginning. Scope became a problem when trying to force the idea of workflow into the process. The workflow should have been handled outside of saving the data to git or any backend. Hence, the workflow should have been handled by each of the build processes of the SSG/Frameworks using the data stored in the markdown. Easy enough to create a field to hold the publish status of the items in a collection. This would have been successful with a roles model for saving data using rules.

Most CMS's use a database for the workflow of the data and publish status. SSG's have one job, build the site using the content defined. Over complicating the workflow to make it fit into a git source control workflow doesn't seem to fit and has caused too many issues in my opinion. Let processes drive the publishing workflow and keep the CMS out of it other than flipping the switches.

**Reply by leo-petrucci** [reply]
I really feel this. At its core Netlify CMS should've stayed simple for ease of maintenance:

- Create or update Markdown files
- Good editing experience
- Easy to customise editor

That's really all that was needed.

**Reply by CamilleScholtz** [reply]
Not to shill or anything (I don't have any connection to the project), but a new CMS called TinaCMS will soon support static sites made with Hugo and such: https://github.com/tinacms/tinacms/discussions/2215. Might be interesting for people that want to use something that is maintained.

**Reply by leo-petrucci** [reply]
> Not to shill or anything (I don't have any connection to the project), but a new CMS called TinaCMS will soon support static sites made with Hugo and such: [tinacms/tinacms#2215](https://github.com/tinacms/tinacms/discussions/2215). Might be interesting for people that want to use something that is maintained.

I just checked this out, turns out if you want to use their editor in production you need to use their cloud solution. :/

**Reply by jimafisk** [reply]
Being a total shill for my own project, we've built an open source, git-backed CMS that does not require a cloud solution to run: https://github.com/plentico/plenti

Things to consider:
- It _only_ supports a GitLab backed currently, although we'll [add GitHub](https://github.com/plentico/plenti/issues/188) as soon as they [support PCKE](https://github.com/orgs/community/discussions/15752)
- It's tightly integrated with our own Svelte-based SSG (it's all one repo), so it won't work with Hugo, Gatsby, etc.

I keep a [video devlog](https://www.youtube.com/watch?v=zPL8xrS9bVg&list=PLbWvcwWtuDm3vNn5ANzgVjyL1YOICT0jE) if you want to see it in action without wasting time spinning things up, although that's pretty quick too :)

**Reply by jhvanderschee** [reply]
I have build something similar to what @jimafisk is saying, but it is closed-source and Hugo only. You can find it here:
https://cms.usecue.com.

**Comment by KaneFreeman** [comment]
As part of another project I started a fork to incorporate some changes I really wanted to see in the app. At first with the project I tried to use NetlifyCMS straight up but kept running into limiting factors for what I was trying to achieve. 

- The markdown editor is very limiting (no table support 😢 )
- The editorial workflow was way more trouble then it was worth (I think I turned it off after just 15 minutes of having it on)
- Bundling into a TypeScript react had issues due to the published types not matching the underlying code
- Styling preview templates from your main app difficult due to the preview being in a frame (especially if you are using something like styled components).
- The dependencies of the app are getting rather out of date, making bundling them into a more modern app rather hard

All of this AND the severe lack of activity in this repo led me to start my own fork. I am planning on dedicating quite a lot of time to it as I go through the process of simplifying it (removing the editorial workflow for one) and modernizing it.

I would love to get input from anyone here on the direction I take. I am still working on some of the infrastructure pieces (which should be in place in the next week or so).

**Repo**: https://github.com/StaticJsCMS/static-cms
**Roadmap**: https://github.com/StaticJsCMS/static-cms/issues/2
**Discussion**: https://github.com/StaticJsCMS/static-cms/discussions/4
**NPM Package**: https://www.npmjs.com/package/@staticcms/core
**Docs**: https://staticcms.org/

**Comment by martinjagodic** [comment, marked as answer]
Great news :grin: The project is not dead anymore! We are honoured to continue the path of Netlify CMS as Decap CMS.

Read more at [Netlify blog](https://www.netlify.com/blog/netlify-cms-to-become-decap-cms/) or [Decap blog](https://decapcms.org/blog/2023/02/introducing-decap)

**Comment by jhvanderschee** [comment]
WOW that is really great news!!!!!

Some thoughts: https://www.usecue.com/blog/the-future-of-netlify-cms/

**Comment by craftzneko** [comment]
Was literally researching options to replace netlify CMS today and i see this news! Amazing 👍

**Comment by kyoshino** [comment]
Hi everyone 👋 The Decap news came as a surprise two weeks ago, but I thought my project was still worth sharing in this thread:

I have recently started rebuilding Netlify CMS from scratch and migrating my clients to the unofficial Svelte port to fix some UX and i18n issues. The resulting product basically materializes what **Netlify CMS v3** would look like, such as a [modern UI](https://github.com/decaporg/decap-cms/issues/2557) with [dark mode](https://github.com/decaporg/decap-cms/issues/3267), an all-new media library, [stock photo integrations](https://github.com/decaporg/decap-cms/issues/2579), [custom collection icons](https://github.com/decaporg/decap-cms/issues/1040), etc., while ensuring basic compatibility with Netlify CMS v2.

Check out **Sveltia CMS** today 👉 https://github.com/sveltia/sveltia-cms

It’s still early beta with no test coverage but open source. Hope some of you will like it!
